vdn/server.go

```diff
@@ -136,9 +124,65 @@ func (s *Server) Start() {
 		s.host.ConnManager().Protect(p.ID, "bootnode")
 		s.bootPeerInfo[p.ID] = p
 	}
+
+	dhtOpts := []dht.Option{
+		dht.Mode(dht.ModeServer),
+		dht.ProtocolPrefix("/bsc/validator/disc"),
+	}
+
+	var err error
+	s.dht, err = dht.New(s.ctx, s.host, dhtOpts...)
+	if err != nil {
+		log.Error("Failed to create DHT", "err", err)
+		return
+	}
+
+	// Now bootstrap the DHT, which also starts its background refresh loop.
 	if err := s.dht.Bootstrap(s.ctx); err != nil {
-		log.Error("Kad-DHT bootstrap failed", "err", err)
+		log.Error("DHT bootstrap failed", "err", err)
+		return
 	}
+
+	// Set up the discovery interface
+	routingDiscovery := drouting.NewRoutingDiscovery(s.dht)
+	// Advertise ourselves under the Rendezvous string
+	log.Info("Advertising Rendezvous", "topic", Rendezvous)
+	dutil.Advertise(s.ctx, routingDiscovery, Rendezvous)
```

Comment:
I think your implementation of Rendezvous is not centralized. Right? It does not relay on a central point, because of every node will Advertise?

Reply:
Does Rendezvous guarantee that all nodes will be discovered eventually?